tasks/eslint.js

```diff
@@ -0,0 +1,42 @@
+import { CLIEngine } from 'eslint';
+
+const OPTION_DEFAULTS = {
+  paths: null,
+  cache: null,
+  fix: false
+};
+
+module.exports = grunt => {
+  grunt.registerMultiTask('eslint',function () {
+    const options = this.options(OPTION_DEFAULTS);
+
+    if (!options.paths) {
+      grunt.fatal(new Error('No eslint.options.paths specified'));
+      return;
+    }
+
+    const cli = new CLIEngine({
+      cache: options.cache,
+      fix: options.fix,
+      cwd: grunt.config.get('root'),
+    });
+
+    const report = cli.executeOnFiles(options.paths);
+
+    // output fixes to disk
+    if (options.fix) {
+      CLIEngine.outputFixes(report);
+    }
+
+    // log the formatted linting report
+    const formatter = cli.getFormatter();
+
+    const errTypes = [];
+    if (report.errorCount > 0) errTypes.push('errors');
+    if (report.warningCount > 0) errTypes.push('warning');
+    if (!errTypes.length) return;
+
+    grunt.log.write(formatter(report.results));
```

Comment:
Do we get to this point if there are no errors? If there are no errors, do we still want to see the report results? Without knowing what the report contains, it seems like this _could_ be useful information?